### PR TITLE
Fix collocation check on listSnapshots method

### PR DESCRIPTION
**Pull request by averzicco**
Fix the issue #1046
Basically will allow to retrieve snapshots also from reaper instances which are configured also with `DatacenterAvailability` = `ALL` (the only one excluded with the collocation check)

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java [hunk 1]
```diff
@@ -698,8 +698,7 @@ public boolean nodeIsAccessibleThroughJmx(String nodeDc, String node) {
    */
   public List<Snapshot> listSnapshots(Node host) throws ReaperException {
     try {
-      if (context.config.getDatacenterAvailability().isInCollocatedMode()
-          && context.jmxConnectionFactory.getHostConnectionCounters().getSuccessfulConnections(
+      if (context.jmxConnectionFactory.getHostConnectionCounters().getSuccessfulConnections(
```
Comment (adejanovski) [inline src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java:701]:
```suggestion
      if (!context.config.getDatacenterAvailability().isInCollocatedMode()
          && context.jmxConnectionFactory.getHostConnectionCounters().getSuccessfulConnections(
```

The check was indeed wrong as we need to make sure we're not in collocated (distributed) mode for the snapshots to be listed.
Listing snapshots involves a direct call to each node in the cluster which is most probably impossible in collocated mode.

**Review by adejanovski — CHANGES_REQUESTED** [review]
Hi @averzicco,

thanks for pushing this PR!
As mentioned in my comment, there's a bug indeed and the condition should pass only if we're **not** in collocated mode.

File: src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java (hunk 1 above)
Comment (averzicco) [reply src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java:701]: Hi @adejanovski 

seems fair. I've applied the change you suggested.

Thanks!

**Comment by averzicco** [comment]
hi @adejanovski 

it seems that integration tests are failing because executed with `datacenterAvailability: LOCAL` but expect to be able to list snapshots.
Should I modify the integration test? 
Wouldn't this cause trouble to whom is using reaper right now with DC availability LOCAL, SIDECAR or EACH and snapshots are "working"?

**Comment by adejanovski** [comment]
> it seems that integration tests are failing because executed with datacenterAvailability: LOCAL but expect to be able to list snapshots.
Should I modify the integration test?

Yes, currently they're validating the wrong behavior so they indeed need to be modified. We'll need to introduce a new test type "all" and introduce a new cucumber tag so that we can avoid running the snapshot test for other test types.
This is going to be a non trivial change, so let me know if you need me to provide more guidance.

> Wouldn't this cause trouble to whom is using reaper right now with DC availability LOCAL, SIDECAR or EACH and snapshots are "working"?

None of these settings should allow access to all nodes from all reaper instances, which gives two possibilities:

- It looks like it's working but it's only showing partial results
- Reaper should be configured with ALL instead

Either way, I'm fine with moving forward with the change.

Thanks!

**Comment by averzicco** [comment]
what if we allow listing snapshots for all values of availability and warn the user on UI about partial results?

**Comment by averzicco** [comment]
@adejanovski any inputs on this?

> what if we allow listing snapshots for all values of availability and warn the user on UI about partial results?

**Comment by adejanovski** [comment]
> @adejanovski any inputs on this?
> 
> > what if we allow listing snapshots for all values of availability and warn the user on UI about partial results?

It's more involved as a change and requires some work on the frontend side, but if you manage to make this work and not confusing for users, then wfm 👍

**Comment by averzicco** [comment]
@adejanovski added an endpoint used to retrieve the value of datacenterAvailability configuration parameter and modified the components which list snapshots to show the warning message `Reaper is configured with datacenterAvailability={datacenterAvailability}, only snapshots on reachable nodes are listed`. 

Not sure if this is confusing for users, what do you think?

PS. this is my first time dealing with react, please check if what I did makes sense

Snapshot Menu
Snapshot Menu

Node detail -> snapshot tab
Node detail

**Review by adejanovski — APPROVED** [review]
LGTM ✅ 

Thanks @averzicco !

**Pull request merged**